src/inmanta/agent/handler.py

```diff
@@ -483,6 +542,68 @@ def get_client(self) -> protocol.SessionClient:
             self._client = protocol.SessionClient("agent", self._agent.sessionid)
         return self._client
 
+    def get_file(self, hash_id: str) -> Optional[bytes]:
+        """
+        Retrieve a file from the fileserver identified with the given id. The convention is to use the sha1sum of the
+        content to identify it.
+
+        :param hash_id: The id of the content/file to retrieve from the server.
+        :return: The content in the form of a bytestring or none is the content does not exist.
+        """
+
+        def call() -> typing.Awaitable[Result]:
+            return self.get_client().get_file(hash_id)
+
+        result = self.run_sync(call)
+        if result.code == 404:
+            return None
+        elif result.result and result.code == 200:
+            file_contents = base64.b64decode(result.result["content"])
+            actual_hash_of_file = hash_file(file_contents)
+            if hash_id != actual_hash_of_file:
+                raise Exception(f"File hash verification failed, expected: {hash_id} but got {actual_hash_of_file}")
+            return file_contents
+        else:
+            raise Exception("An error occurred while retrieving file %s" % hash_id)
+
+    def stat_file(self, hash_id: str) -> bool:
+        """
+        Check if a file exists on the server. This method does and async call to the server and blocks on the result.
```

Comment:
I don't really see the relevance of the second sentence. @arnaudsjs @wouterdb do you?

Reply:
Me neither.